cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterRebalanceAssemblyOperatorTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.assembly;
+
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.strimzi.api.kafka.KafkaClusterRebalanceList;
+import io.strimzi.api.kafka.model.DoneableKafkaClusterRebalance;
+import io.strimzi.api.kafka.model.KafkaBridge;
+import io.strimzi.api.kafka.model.KafkaClusterRebalance;
+import io.strimzi.api.kafka.model.KafkaClusterRebalanceBuilder;
+import io.strimzi.api.kafka.model.KafkaClusterRebalanceSpec;
+import io.strimzi.api.kafka.model.KafkaClusterRebalanceSpecBuilder;
+import io.strimzi.api.kafka.model.status.Condition;
+import io.strimzi.operator.KubernetesVersion;
+import io.strimzi.operator.PlatformFeaturesAvailability;
+import io.strimzi.operator.cluster.ResourceUtils;
+import io.strimzi.operator.cluster.model.CruiseControl;
+import io.strimzi.operator.cluster.operator.assembly.cruisecontrol.MockCruiseControl;
+import io.strimzi.operator.cluster.operator.resource.ResourceOperatorSupplier;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.model.Labels;
+import io.strimzi.operator.common.operator.resource.CrdOperator;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.junit5.Checkpoint;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockserver.integration.ClientAndServer;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(VertxExtension.class)
+public class KafkaClusterRebalanceAssemblyOperatorTest {
+
+    private static final String HOST = "localhost";
+    private static final String CLUSTER_NAMESPACE = "cruise-control-namespace";
+    private static final String CLUSTER_NAME = "cruise-control-test-cluster";
+
+    private final KubernetesVersion kubernetesVersion = KubernetesVersion.V1_11;
+
+    private static final Logger log = LogManager.getLogger(KafkaClusterRebalanceAssemblyOperatorTest.class.getName());
+
+    private static ClientAndServer ccServer;
+
+    @BeforeAll
+    public static void before() throws IOException, URISyntaxException {
+        ccServer = MockCruiseControl.getCCServer(CruiseControl.REST_API_PORT, 0);
```

Comment:
we should try to make the `pendingCalls` parameter configurable on test basis because we want to have different tests which provides the result immediately or after X calls. This is just a comment we should think about, nothing to address right now.

Reply:
@tomncooper because this is all about mocking, does it work if we make the various `setupXXXX` methods to be public in the `MockCruiseControl` class?
I mean, if there is a specific test where I need a `pendingCalls` greater than zero, we could just call the `setupCCUserTasksResponse(ccServer, pendingCalls);` in the specific test to change the behavior. Wdyt?

Reply:
Good idea